src/main/java/seedu/address/logic/commands/deletecommand/DeleteGeneralEventCommand.java

```diff
@@ -0,0 +1,51 @@
+package seedu.address.logic.commands.deletecommand;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.event.GeneralEvent;
+
+/**
+ * Deletes a person identified using it's displayed index from the remindMe.
+ */
+public class DeleteGeneralEventCommand extends DeleteCommand {
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the event identified by the index number used in the displayed general event list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_GENERAL_EVENT_SUCCESS = "Deleted General Event: %1$s";
+
+    private final Index targetIndex;
+
+    public DeleteGeneralEventCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<GeneralEvent> lastShownList = model.getFilteredEventList();
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
```

Comment:
I believe that Model actually has a hasEvent(int index) method that can be used here instead!